be/tool/beck.c

```diff
@@ -534,104 +552,122 @@ int main(int argc, char **argv)
+	if (!beck_scanner.s_print_invalid_oids)
+		printf("Will not print INVALID GOB IDs if found since '-e'"
+		       "option was not specified. \n");
+
+	if (beck_builder.b_dom_path == NULL && !dry_run && !print_gen_id)
 		errx(EX_USAGE, "Specify domain path (-d).");
 	if (spath != NULL) {
 		sfd = open(spath, O_RDONLY);
 		if (sfd == -1)
 			err(EX_NOINPUT, "Cannot open snapshot \"%s\".", spath);
 	}
-	s.s_file = fdopen(sfd, "r");
-	if (s.s_file == NULL)
+	beck_scanner.s_file = fdopen(sfd, "r");
+	if (beck_scanner.s_file == NULL)
 		err(EX_NOINPUT, "Cannot open snapshot.");
-	if (s.s_size == 0) {
-		result = fseek(s.s_file, 0, SEEK_END);
+	if (beck_scanner.s_size == 0) {
+		result = fseek(beck_scanner.s_file, 0, SEEK_END);
 		if (result != 0)
 			err(EX_NOINPUT, "Cannot seek snapshot to the end.");
-		b.b_size = s.s_size = ftello(s.s_file);
-		if (s.s_size == -1)
+		beck_builder.b_size = beck_scanner.s_size =
+						ftello(beck_scanner.s_file);
+		if (beck_scanner.s_size == -1)
 			err(EX_NOINPUT, "Cannot tell snapshot size.");
-		result = fseek(s.s_file, 0, SEEK_SET);
+		result = fseek(beck_scanner.s_file, 0, SEEK_SET);
 		if (result != 0)
 		      err(EX_NOINPUT, "Cannot seek snapshot to the beginning.");
-		printf("Snapshot size: %"PRId64".\n", s.s_size);
+		printf("Snapshot size: %"PRId64".\n", beck_scanner.s_size);
 	}
 
-	if (b.b_be_config_file && !dry_run && !print_gen_id) {
-		fp = fopen(b.b_be_config_file, "r");
+	if (beck_builder.b_be_config_file && !dry_run && !print_gen_id) {
+		fp = fopen(beck_builder.b_be_config_file, "r");
 		if (fp == NULL)
 			err(EX_NOINPUT, "Failed to open yaml file \"%s\".",
-			    b.b_be_config_file);
+			    beck_builder.b_be_config_file);
 		fclose(fp);
 	}
-	generation_id_get(s.s_file, &gen_id);
+	generation_id_get(beck_scanner.s_file, &gen_id);
 	if (print_gen_id) {
 		generation_id_print(gen_id);
 		printf("\n");
-		fclose(s.s_file);
+		fclose(beck_scanner.s_file);
 		return EX_OK;
 	}
 	if (gen_id != 0)
-		s.s_gen_found = true;
-	if (s.s_gen != 0) {
+		beck_scanner.s_gen_found = true;
+	if (beck_scanner.s_gen != 0) {
 		printf("\nReceived source segment header generation id\n");
-		generation_id_print(s.s_gen);
+		generation_id_print(beck_scanner.s_gen);
 		printf("\n");
 		fflush(stdout);
 	}
 
-	s.s_gen = gen_id ?: s.s_gen;
+	beck_scanner.s_gen = gen_id ?: beck_scanner.s_gen;
 	printf("\nSource segment header generation id to be used by beck.\n");
-	generation_id_print(s.s_gen);
+	generation_id_print(beck_scanner.s_gen);
 	printf("\n");
 	fflush(stdout);
 	/*
 	 *  If both segment's generation identifier is corrupted then abort
 	 *  beck tool.
 	 */
-	if (!dry_run && s.s_gen == 0) {
+	if (!dry_run && beck_scanner.s_gen == 0) {
 		printf("Cannot find any segment header generation identifer");
 		return EX_DATAERR;
 	}
+	qinit(&beck_scanner.s_bnode_q, MAX_SCAN_QUEUED);
+	result = M0_THREAD_INIT(&beck_scanner.s_thread, struct scanner *,
+				NULL, &scanner_thread, &beck_scanner, "scannner");
```

Comment:
Please check if this line exceeds 80 chars.

Reply:
Yes it was. Thanks for pointing it.